File: modules/apps/object/object-admin-rest-client-js/src/models/ObjectLayout.ts
/**
 * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

			import {ObjectLayoutTab} from './ObjectLayoutTab';

/**
 * @author Javier Gamarra
 * @generated
 */

	export class ObjectLayout {
			"actions"?: {[key: string]: {[key: string]: string;};};
			"dateCreated"?: Date;
			"dateModified"?: Date;
			"defaultObjectLayout"?: boolean;
			"id"?: number;
			"name"?: {[key: string]: string;};
			"objectDefinitionExternalReferenceCode"?: string;
			"objectDefinitionId"?: number;
			"objectLayoutTabs"?: Array<ObjectLayoutTab>;

		static "discriminator": string | undefined = undefined;

	static "attributeTypeMap": Array<{
		baseName: string;
		name: string;
		type: string;
	}> = [
		{
			baseName: "actions",
			name: "actions",
			type: "{[key: string]: {[key: string]: string;};}",
		},
		{
			baseName: "dateCreated",
			name: "dateCreated",
			type: "Date",
		},
		{
			baseName: "dateModified",
			name: "dateModified",
			type: "Date",
		},
		{
			baseName: "defaultObjectLayout",
			name: "defaultObjectLayout",
			type: "boolean",
		},
		{
			baseName: "id",
			name: "id",
			type: "number",
		},
		{
			baseName: "name",
			name: "name",
			type: "{[key: string]: string;}",
		},
		{
			baseName: "objectDefinitionExternalReferenceCode",
			name: "objectDefinitionExternalReferenceCode",
			type: "string",
		},
		{
			baseName: "objectDefinitionId",
			name: "objectDefinitionId",
			type: "number",
		},
		{
			baseName: "objectLayoutTabs",
			name: "objectLayoutTabs",
			type: "Array<ObjectLayoutTab>",
		},
		];

		static getAttributeTypeMap() {
				return ObjectLayout.attributeTypeMap;
		}
	}

File: modules/apps/object/object-api/src/main/java/com/liferay/object/model/ObjectLayout.java
/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.object.model;

import com.liferay.portal.kernel.annotation.ImplementationClassName;
import com.liferay.portal.kernel.model.PersistedModel;
import com.liferay.portal.kernel.util.Accessor;

import org.osgi.annotation.versioning.ProviderType;

/**
 * The extended model interface for the ObjectLayout service. Represents a row in the &quot;ObjectLayout&quot; database table, with each column mapped to a property of this class.
 *
 * @author Marco Leo
 * @see ObjectLayoutModel
 * @generated
 */
@ImplementationClassName("com.liferay.object.model.impl.ObjectLayoutImpl")
@ProviderType
public interface ObjectLayout extends ObjectLayoutModel, PersistedModel {

	/*
	 * NOTE FOR DEVELOPERS:
	 *
	 * Never modify this interface directly. Add methods to <code>com.liferay.object.model.impl.ObjectLayoutImpl</code> and rerun ServiceBuilder to automatically copy the method declarations to this interface.
	 */
	public static final Accessor<ObjectLayout, Long> OBJECT_LAYOUT_ID_ACCESSOR =
		new Accessor<ObjectLayout, Long>() {

			@Override
			public Long get(ObjectLayout objectLayout) {
				return objectLayout.getObjectLayoutId();
			}

			@Override
			public Class<Long> getAttributeClass() {
				return Long.class;
			}

			@Override
			public Class<ObjectLayout> getTypeClass() {
				return ObjectLayout.class;
			}

		};

	public java.util.List<ObjectLayoutTab> getObjectLayoutTabs();

	public void setObjectLayoutTabs(
		java.util.List<ObjectLayoutTab> objectLayoutTabs);

}
// SB-Hash:-140128237